domain-models-runtime/src/test/java/org/folio/rest/persist/PostgresClientIT.java

```diff
@@ -643,6 +648,33 @@ public void updateSectionSingleQuote(TestContext context) {
     }));
   }
 
+  @Test
+  public void selectWithTimeout(TestContext context) {
+      PostgresClient client = postgresClient();
+      client.getSQLConnection(2500, conn -> {
+        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertSuccess());
+        client.selectSingle(conn, "SELECT 1, pg_sleep(2);", context.asyncAssertFailure());
```

Comment:
Only one query can run on conn at the same time, for parallel queries a second conn is needed.
Please check the SQLSTATE of the exception returned.